xchange-blockchain/src/main/java/org/knowm/xchange/blockchain/BlockchainAdapters.java

```diff
@@ -86,4 +108,144 @@ public static CurrencyPair toCurrencyPairBySymbol(BlockchainSymbol blockchainSym
+                .side(Order.OrderType.BID.equals(stopOrder.getType())? BUY.toUpperCase() : SELL.toUpperCase())
+                .orderQty(stopOrder.getOriginalAmount())
+                .price(stopOrder.getLimitPrice())
+                .stopPx(stopOrder.getStopPrice())
+                .clOrdId(generateClOrdId())
+                .build();
+    }
+
+    private static String generateClOrdId() {
+        String uuid = UUID.randomUUID().toString();
+        uuid = uuid.substring(0, 16).replace("-", "");
+        return uuid;
+    }
+
+    public static UserTrades toUserTrades(List<BlockchainOrder> blockchainTrades) {
+        List<UserTrade> trades = blockchainTrades.stream()
+                .map(blockchainTrade -> new UserTrade.Builder()
+                                .type(blockchainTrade.getOrderType())
+                                .originalAmount(blockchainTrade.getCumQty())
+                                .currencyPair(blockchainTrade.getSymbol())
+                                .price(blockchainTrade.getPrice())
+                                .timestamp(blockchainTrade.getTimestamp())
+                                .id(Long.toString(blockchainTrade.getExOrdId()))
+                                .orderId(blockchainTrade.getClOrdId())
+                                .build()
+                ).collect(Collectors.toList());
+        Long lastId = blockchainTrades.stream().map(BlockchainOrder::getExOrdId).max(Long::compareTo).orElse(0L);
+        return new UserTrades(trades, lastId, Trades.TradeSortType.SortByTimestamp);
+    }
+
+   public static ExchangeMetaData adaptMetaData(Map<String, BlockchainSymbol> markets) {
+       Map<CurrencyPair, CurrencyPairMetaData> currencyPairs = new HashMap<>();
+       Map<Currency, CurrencyMetaData> currency = new HashMap<>();
+
+       for (Map.Entry<String, BlockchainSymbol> entry : markets.entrySet()) {
+           CurrencyPair pair = BlockchainAdapters.toCurrencyPairBySymbol(entry.getValue());
+           BigDecimal minScale = BigDecimal.valueOf(Math.pow(10, (entry.getValue().getMinOrderSizeScale())*-1));
+           BigDecimal minAmount = entry.getValue().getMinOrderSize().multiply(minScale);
+           BigDecimal maxScale = BigDecimal.valueOf(Math.pow(10, (entry.getValue().getMaxOrderSizeScale())*-1));
+           BigDecimal maxAmount = entry.getValue().getMaxOrderSize().multiply(maxScale);
+           CurrencyPairMetaData currencyPairMetaData =
+                   new CurrencyPairMetaData.Builder()
+                           .baseScale(entry.getValue().getBaseCurrencyScale())
+                           .priceScale(entry.getValue().getCounterCurrencyScale())
+                           .minimumAmount(minAmount)
+                           .maximumAmount(maxAmount)
+                           .build();
+           currencyPairs.put(pair, currencyPairMetaData);
+           currency.put(entry.getValue().getBaseCurrency(), null);
```

Comment:
why null?

Reply:
There should be the currencyMetadata `CurrencyMetaData(Integer scale, BigDecimal withdrawalFee)` of which we only have the scale and no other data .... so I left it in null.
But I can build the currency with only the scale.